plugins/by-name/neorg/default.nix

```diff
@@ -1,173 +1,249 @@
 {
   lib,
-  helpers,
   config,
   pkgs,
   ...
 }:
 let
-  cfg = config.plugins.neorg;
+  inherit (lib.nixvim)
+    defaultNullOpts
+    mkNullOrStr
+    mkNullOrOption
+    mkNullOrOption'
+    ;
+  inherit (lib) types;
 in
-with lib;
-{
-  options.plugins.neorg = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "neorg";
-
-    package = lib.mkPackageOption pkgs "neorg" {
-      default = [
-        "vimPlugins"
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "neorg";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  # TODO introduced 2024-12-12: remove after 25.05
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "lazyLoading"
+    [
+      "logger"
+      "plugin"
+    ]
+    [
+      "logger"
+      "useConsole"
+    ]
+    [
+      "logger"
+      "highlights"
+    ]
+    [
+      "logger"
+      "useFile"
+    ]
+    [
+      "logger"
+      "level"
+    ]
+    [
+      "logger"
+      "floatPrecision"
+    ]
+  ];
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
         "neorg"
       ];
+    in
+    [
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "modules" ]) (
+        basePluginPath
+        ++ [
+          "settings"
+          "load"
+        ]
+      ))
+      (lib.mkRemovedOptionModule (
+        basePluginPath
+        ++ [
+          "logger"
+          "modes"
+        ]
+      ) "Please use `settings.logger.modes` but you now need to provide a list of submodules.")
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "neorgTelescopePackage" ]) (
+        basePluginPath
+        ++ [
+          "telescopeIntegration"
+          "package"
+        ]
+      ))
+    ];
+
+  extraOptions = {
+    telescopeIntegration = {
```

Comment:
If we're already creating a bunch of renames maybe follow suite of the others `enableTelescope`?